functions/go/set-gcp-resource-ids/main.go

```diff
@@ -10,55 +10,32 @@ import (
 	"k8s.io/apimachinery/pkg/runtime/schema"
 )
 
+var (
+	Folder = schema.GroupVersionKind{Group: "resourcemanager.cnrm.cloud.google.com", Version: "v1beta1", Kind: "Folder"}
+	GCPProject = schema.GroupVersionKind{Group: "resourcemanager.cnrm.cloud.google.com", Version: "v1beta1", Kind: "Project"}
+	ConfigControllerContext = schema.GroupVersionKind{Group: "core.cnrm.cloud.google.com", Version: "v1beta1", Kind: "ConfigConnectorContext"}
+	)
+
 func main() {
 	// TODO: fn.AsMain should support an "easy mode" where it runs against a directory
-	if err := fn.AsMain(fn.ResourceListProcessorFunc(Run)); err != nil {
+	if err := fn.AsMain(&SetGCPProject{}); err != nil {
 		os.Exit(1)
 	}
 }
 
+var _ fn.Runner = &SetGCPProject{}
+
 type SetGCPProject struct {
 	ProjectID string `json:"projectID,omitempty"`
 }
 
-func Run(rl *fn.ResourceList) (bool, error) {
-	f := SetGCPProject{}
-
-	err := f.LoadConfig(rl.FunctionConfig)
-	if err != nil {
-		rl.Results = append(rl.Results, fn.ErrorConfigObjectResult(fmt.Errorf("functionConfig error: %w", err), rl.FunctionConfig))
-		return true, nil
-	}
-
-	if err := f.Transform(rl.Items); err != nil {
-		rl.Results = append(rl.Results, fn.ErrorResult(err))
-	}
-	return true, nil
-}
```

Comment:
`fnRunner` encapsulates the `LoadConfig` and `Transform` logics.

Reply:
Wonderful :tada: - thank you!